src/Security/Authenticator/GitHubAuthenticator.php

```diff
@@ -80,11 +81,11 @@ public function getUser($credentials, UserProviderInterface $userProvider): ?Use
             return $existingUser;
         }
 
-        /** @var User $user */
+        /** @var User|null $user */
         $user = $this->em->getRepository(User::class)
             ->findOneBy(['email' => $githubUser->getEmail()]);
 
-        if ($user) {
+        if ($user instanceof User) {
```

Comment:
I hate this kind of change. Previous code was totally valid.
Is it mandatory?

Reply:
I added it because PHPstan was saying the following code was unreachable, but I removed it and it works fine now.